packages/screens/WebView/BrowserDetail.tsx

```diff
@@ -0,0 +1,329 @@
+  layout,
+} from "@/utils/style/layout";
+
+function BrowserHeader({
+  url,
+  shareLink,
+}: {
+  url: string;
+  shareLink: () => void;
+}) {
+  const navigation = useAppNavigation();
+  const { setToast } = useFeedbacks();
+
+  async function openInBrowser() {
+    try {
+      const supported = await Linking.canOpenURL(url);
+
+      if (supported) {
+        Linking.openURL(url);
+      } else {
+        setToast({
+          mode: "mini",
+          type: "error",
+          variant: "solid",
+          message: "Error: Link is not valid",
+        });
+      }
+    } catch (error) {
+      setToast({
+        mode: "mini",
+        type: "error",
+        variant: "solid",
+        message: "Error: " + error,
+      });
+    }
+  }
+  return (
+    <View
+      style={{
+        backgroundColor: neutral00,
+        flexDirection: "row",
+        justifyContent: "space-between",
+        height: MOBILE_HEADER_HEIGHT,
+        maxHeight: MOBILE_HEADER_HEIGHT,
+        width: "100%",
+        alignItems: "center",
+        paddingHorizontal: layout.spacing_x2,
+        position: "absolute",
+        top: 0,
+        zIndex: 9999,
+        borderBottomWidth: 1,
+        borderColor: neutral22,
+      }}
+    >
+      <View
+        style={{
+          flex: 1,
+          alignItems: "center",
+          justifyContent: "space-between",
+          flexDirection: "row",
+        }}
+      >
+        <CustomPressable onPress={() => navigation.goBack()}>
+          <SVG source={closeSVG} width={28} height={28} />
+        </CustomPressable>
+        <View style={{ flex: 1, alignItems: "center" }}>
+          <BrandText
+            style={[fontSemibold14, { marginTop: 8, marginBottom: -10 }]}
+          >
+            Popular Collections
+          </BrandText>
+          <View
+            style={{
+              flexDirection: "row",
+              alignItems: "center",
+              gap: 4,
+              flex: 1,
+            }}
+          >
+            <SVG source={lockSVG} width={12} height={12} />
+            <BrandText style={[fontSemibold13, { color: neutralA3 }]}>
+              {url.length > 40 ? url.substring(0, 30) + "..." : url}
+            </BrandText>
+          </View>
+        </View>
+        <DropdownWithListItem
+          items={[
+            {
+              onPress: openInBrowser,
+              name: "Open in browser",
+              icon: browseSVG,
+            },
+            {
+              onPress: shareLink,
+              name: "Share the link",
+              icon: shareSVG,
+            },
+          ]}
+          iconSize={34}
+          icon={dotSVG}
+        />
+      </View>
+    </View>
+  );
+}
+
+function BrowserFooter({
+  browserNav,
+  webViewRef,
+  shareLink,
+}: {
+  browserNav: BrowserNavigationType;
+  webViewRef: WebView | null;
+  shareLink: () => void;
+}) {
+  const bottomTabs = [
+    {
+      name: "back",
+      icon: backSVG,
+      onPress: () => webViewRef?.goBack(),
```

Comment:
why sometimes is null? can you explain when is that case?

Reply:
https://react.dev/learn/referencing-values-with-refs#refs-and-the-dom:~:text=When%20you%20pass%20a%20ref%20to%20a%20ref%20attribute%20in%20JSX%2C%20like%20%3Cdiv%20ref%3D%7BmyRef%7D%3E%2C%20React%20will%20put%20the%20corresponding%20DOM%20element%20into%20myRef.current.%20Once%20the%20element%20is%20removed%20from%20the%20DOM%2C%20React%20will%20update%20myRef.current%20to%20be%20null.%20You%20can%20read%20more%20about%20this%20in%20Manipulating%20the%20DOM%20with%20Refs.